signer/src/transaction_signer.rs

```diff
@@ -512,6 +512,11 @@ where
             WstsNetMessage::DkgBegin(_) => {
                 tracing::info!("handling DkgBegin");
 
+                if chain_tip_report.chain_tip_status != ChainTipStatus::Canonical {
+                    tracing::warn!("received wsts message for a different chain tip");
+                    return Ok(());
+                }
+
                 if !chain_tip_report.sender_is_coordinator {
                     tracing::warn!("received coordinator message from non-coordinator signer");
                     return Ok(());
```

Comment:
We should probably add a function in the `MsgChainTipReport` that implements this logic, since it appears to be the same throughout.